amaranth/sim/_pyrtl.py

```diff
@@ -387,8 +415,47 @@ def on_Switch(self, stmt):
             with self.emitter.indent():
                 self(stmts)
 
+    def emit_format(self, format):
+        format_string = []
+        args = []
+        for chunk in format._chunks:
+            if isinstance(chunk, str):
+                format_string.append(chunk.replace("{", "{{").replace("}", "}}"))
+            else:
+                value, format_desc = chunk
+                value = self.rhs.sign(value)
+                if format_desc.endswith("s"):
+                    format_desc = format_desc[:-1]
+                    value = f"value_to_string({value})"
+                format_string.append(f"{{:{format_desc}}}")
+                args.append(value)
+        format_string = "".join(format_string)
+        args = ", ".join(args)
+        return f"{format_string!r}.format({args})"
+
+    def on_Print(self, stmt):
+        self.emitter.append(f"print({self.emit_format(stmt.message)}, end='', file=state.print_output)")
+
     def on_Property(self, stmt):
-        raise NotImplementedError # :nocov:
+        if stmt.kind == Property.Kind.Cover:
+            if stmt.message is not None:
+                self.emitter.append(f"if {self.rhs.sign(stmt.test)}:")
+                with self.emitter.indent():
+                    fname, line = stmt.src_loc
```

Comment:
(Since someone got confused on Mastodon)

```suggestion
                    filename, line = stmt.src_loc
```